.github/workflows/gui-pull-request.yml

```diff
@@ -0,0 +1,64 @@
+# This file is not auto-generated. Feel free to edit it.
+
+name: ✨ GUI Pull Request
+
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read # Read-only access to repository contents
+  issues: write # Write access to issues
+  pull-requests: write # Write access to pull requests
+  statuses: write # Write access to commit statuses
+  checks: write
+
+jobs:
+  changed-files:
+    runs-on: ubuntu-latest
+    name: 🔍 Detect changed files in GUI
+    outputs:
+      all_changed_files: ${{ steps.changed-files.outputs.all_changed_files }}
+      any_changed: ${{ steps.changed-files.outputs.any_changed }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Again this fetch-depth setting is really bad here. You should limit the depth to `2`, that should be enough to detect changes.